ethers-signers/Cargo.toml

```diff
@@ -23,10 +23,10 @@ coins-bip39 = "0.8.1"
 coins-ledger = { version = "0.7.1", default-features = false, optional = true }
 hex = { version = "0.4.3", default-features = false, features = ["std"] }
 async-trait = { version = "0.1.50", default-features = false }
-elliptic-curve = { version = "0.12.3", default-features = false }
+elliptic-curve = { version = "0.13.2", default-features = false }
 sha2 = { version = "0.10.6", default-features = false }
 rand = { version = "0.8.5", default-features = false }
-yubihsm = { version = "0.41.0", features = ["secp256k1", "http", "usb"], optional = true }
+yubihsm = { git = "https://github.com/iqlusioninc/yubihsm.rs", version = "0.42.0-pre", features = ["secp256k1", "http", "usb"], optional = true }
```

Comment:
TODO: Need proper release, not sure why we're getting 
```
57 |     fn from(signer: YubiSigner<Secp256k1>) -> Self {
   |                                               ^^^^ the trait `yubihsm::asymmetric::signature::DigestSigner<ProxyDigest<CoreWrapper<CtVariableCoreWrapper<Sha256VarCore, elliptic_curve::generic_array::typenum::UInt<elliptic_curve::generic_array::typenum::UInt<elliptic_curve::generic_array::typenum::UInt<elliptic_curve::generic_array::typenum::UInt<elliptic_curve::generic_array::typenum::UInt<elliptic_curve::generic_array::typenum::UInt<UTerm, B1>, B0>, B0>, B0>, B0>, B0>, OidSha256>>>, (yubihsm::ecdsa::Signature<ethers_core::k256::Secp256k1>, RecoveryId)>` is not implemented for `yubihsm::ecdsa::Signer<ethers_core::k256::Secp256k1>`
   |
   = help: the following other types implement trait `yubihsm::asymmetric::signature::DigestSigner<D, S>`:
             <yubihsm::ecdsa::Signer<NistP256> as yubihsm::asymmetric::signature::DigestSigner<D, yubihsm::ecdsa::Signature<NistP256>>>
             <yubihsm::ecdsa::Signer<NistP384> as yubihsm::asymmetric::signature::DigestSigner<D, yubihsm::ecdsa::Signature<NistP384>>>
             <yubihsm::ecdsa::Signer<ethers_core::k256::Secp256k1> as yubihsm::asymmetric::signature::DigestSigner<D, yubihsm::ecdsa::Signature<ethers_core::k256::Secp256k1>>>
note: required by a bound in `Wallet`
  --> ethers-signers/src/wallet/mod.rs:67:22
   |
67 | pub struct Wallet<D: DigestSigner<Sha256Proxy, (RecoverableSignature, RecoveryId)>> {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Wallet`
```

at the moment

Reply:
@gakonst not sure where you're importing `DigestSigner` from, but it might be from `signature` v1 where `yubihsm` (and `k256` are now using `signature` v2?)

Reply:
Also I'll be cutting a release of `yubihsm` soon... probably this week

Reply:
Separately you should be able to replace `DigestSigner<Sha256Proxy, _>` with `PrehashSigner`, although as I say that I realize I need to add the trait impls to `yubihsm` too

Reply:
Roger - yeah that'd be great, once YubiHSM is updated we can bump k256 on ethers and subsequently reth.

Reply:
Roger - yeah that'd be great, once YubiHSM is updated we can bump k256 on ethers and subsequently reth.

Reply:
On PrehashSigner it seems like it does not have a way to return the `RecoveryId`?

Reply:
Basically what I did was:
1. Convert the `DigestSigner<D, Signature>` into `DigestSigner<D, (Signature, RecoveryId)>`
2. This makes `sign_digest` return the Signature & RecoveryID as expected
3. However this breaks YubiHSM because it seems like `DigestSigner<D, (Signature, RecoveryId)>` is not impl'd there
4. It may be worth making `sign_digest_recoverable` available over trait? https://docs.rs/ecdsa/0.16.0/ecdsa/struct.SigningKey.html#method.sign_digest_recoverable

Reply:
> On PrehashSigner it seems like it does not have a way to return the `RecoveryId`?

You can use this type signature for that:

https://docs.rs/ecdsa/latest/ecdsa/struct.SigningKey.html#impl-PrehashSigner%3C(Signature%3CC%3E,+RecoveryId)%3E-for-SigningKey%3CC%3E

I can get it added to `yubihsm` before the next release.

Reply:
Word, sounds good. In that case yeah would love implementations of both `DigestSigner<_, (Signature, RecoveryId)>` and `DigestSigner<_, (Signature, RecoveryId)>` in yubihsm's new release, that's our current blocker. Know you're busy, not sure what your bandwidth is here / timeline on getting that out. Let us know either way^^

Reply:
You can use v0.42.0-pre.0 now: https://crates.io/crates/yubihsm/0.42.0-pre.0

It has support for both `PrehashSigner<(Signature, RecoveryId)>` and `DigestSigner<_, (Signature, RecoveryId)>`:

https://github.com/iqlusioninc/yubihsm.rs/pull/408/files

Reply:
Alright - I think I got it, thanks boss & @agostbiro for implementing